cpp/src/Slice/Parser.h

```diff
@@ -504,73 +485,42 @@ class Container : public virtual SyntaxTreeBase
-                         const std::string&, const std::string&, NodeType = Real);
+    void destroy() override;
+    virtual ModulePtr createModule(const std::string&);
+    virtual ClassDefPtr createClassDef(const std::string&, int, const ClassDefPtr&);
+    virtual ClassDeclPtr createClassDecl(const std::string&);
+    virtual InterfaceDefPtr createInterfaceDef(const std::string&, const InterfaceList&);
+    virtual InterfaceDeclPtr createInterfaceDecl(const std::string&);
+    virtual ExceptionPtr createException(const std::string&, const ExceptionPtr&, NodeType = Real);
+    virtual StructPtr createStruct(const std::string&, NodeType = Real);
+    virtual SequencePtr createSequence(const std::string&, const TypePtr&, const StringList&, NodeType = Real);
+    virtual DictionaryPtr createDictionary(const std::string&, const TypePtr&, const StringList&, const TypePtr&,
+                                           const StringList&, NodeType = Real);
+    virtual EnumPtr createEnum(const std::string&, bool, NodeType = Real);
+    virtual ConstPtr createConst(const std::string, const TypePtr&, const StringList&, const SyntaxTreeBasePtr&,
+                                 const std::string&, const std::string&, NodeType = Real);
     TypeList lookupType(const std::string&, bool = true);
     TypeList lookupTypeNoBuiltin(const std::string&, bool = true, bool = false);
     ContainedList lookupContained(const std::string&, bool = true);
     ExceptionPtr lookupException(const std::string&, bool = true);
-    UnitPtr unit() const;
-    ModuleList modules() const;
-    ClassList classes() const;
-    InterfaceList interfaces() const;
-    ExceptionList exceptions() const;
-    StructList structs() const;
-    SequenceList sequences() const;
-    DictionaryList dictionaries() const;
-    EnumList enums() const;
-    EnumeratorList enumerators() const;
+    // Finds enumerators using the deprecated unscoped enumerators lookup
     EnumeratorList enumerators(const std::string&) const;
-    ConstList consts() const;
-    ContainedList contents() const;
-    bool hasSequences() const;
-    bool hasStructs() const;
-    bool hasExceptions() const;
-    bool hasDictionaries() const;
-    bool hasClassDecls() const;
-    bool hasClassDefs() const;
-    bool hasInterfaceDecls() const;
-    bool hasInterfaceDefs() const;
-    bool hasOnlyClassDecls() const;
-    bool hasOnlyInterfaces() const;
-    bool hasOperations() const; // interfaces or classes with operations
-    bool hasOtherConstructedOrExceptions() const; // Exceptions or constructed types other than classes.
+    virtual ContainedList contents() const = 0;
     bool hasContentsWithMetaData(const std::string&) const;
-    bool hasAsyncOps() const;
-    bool hasContained(Contained::ContainedType) const;
     std::string thisScope() const;
-    void mergeModules();
     void sort();
     void sortContents(bool);
```

Comment:
"dead" function. The only code that calls sortContents is sortContents itself.

Reply:
You're right! Removed `sortContents`, and also our own `sort` method too.